From 4298ccab8f8909ab727d34de52b0b97d9d589b16 Mon Sep 17 00:00:00 2001
From: Dor Askayo <dor.askayo@gmail.com>
Date: Tue, 12 May 2020 22:45:59 +0300
Subject: [PATCH 02/18] clutter/stage: move get_update_time to the backend

This is done in preparation for a following commit that changes when
updates should be performed, but only for the native backend.

https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1241
---
 clutter/clutter/cogl/clutter-stage-cogl.c | 12 ------------
 src/backends/native/meta-stage-native.c   | 12 ++++++++++++
 src/backends/x11/meta-stage-x11.c         | 12 ++++++++++++
 3 files changed, 24 insertions(+), 12 deletions(-)

diff --git a/clutter/clutter/cogl/clutter-stage-cogl.c b/clutter/clutter/cogl/clutter-stage-cogl.c
index be6d3e770..4e0ecd097 100644
--- a/clutter/clutter/cogl/clutter-stage-cogl.c
+++ b/clutter/clutter/cogl/clutter-stage-cogl.c
@@ -235,17 +235,6 @@ clutter_stage_cogl_schedule_update (ClutterStageWindow *stage_window,
     stage_cogl->update_time = stage_cogl->last_update_time + refresh_interval;
 }
 
-static gint64
-clutter_stage_cogl_get_update_time (ClutterStageWindow *stage_window)
-{
-  ClutterStageCogl *stage_cogl = CLUTTER_STAGE_COGL (stage_window);
-
-  if (stage_cogl->pending_swaps)
-    return -1; /* in the future, indefinite */
-
-  return stage_cogl->update_time;
-}
-
 static void
 clutter_stage_cogl_clear_update_time (ClutterStageWindow *stage_window)
 {
@@ -821,7 +810,6 @@ clutter_stage_window_iface_init (ClutterStageWindowInterface *iface)
   iface->show = clutter_stage_cogl_show;
   iface->hide = clutter_stage_cogl_hide;
   iface->schedule_update = clutter_stage_cogl_schedule_update;
-  iface->get_update_time = clutter_stage_cogl_get_update_time;
   iface->clear_update_time = clutter_stage_cogl_clear_update_time;
   iface->can_redraw = clutter_stage_cogl_can_redraw;
   iface->redraw = clutter_stage_cogl_redraw;
diff --git a/src/backends/native/meta-stage-native.c b/src/backends/native/meta-stage-native.c
index 9b9c45ef3..cb69286c7 100644
--- a/src/backends/native/meta-stage-native.c
+++ b/src/backends/native/meta-stage-native.c
@@ -205,6 +205,17 @@ meta_stage_native_finish_frame (ClutterStageWindow *stage_window)
   meta_renderer_native_finish_frame (META_RENDERER_NATIVE (renderer));
 }
 
+static gint64
+meta_stage_native_get_update_time (ClutterStageWindow *stage_window)
+{
+  ClutterStageCogl *stage_cogl = CLUTTER_STAGE_COGL (stage_window);
+
+  if (stage_cogl->pending_swaps)
+    return -1; /* in the future, indefinite */
+
+  return stage_cogl->update_time;
+}
+
 static void
 meta_stage_native_init (MetaStageNative *stage_native)
 {
@@ -227,4 +238,5 @@ clutter_stage_window_iface_init (ClutterStageWindowInterface *iface)
   iface->get_views = meta_stage_native_get_views;
   iface->get_frame_counter = meta_stage_native_get_frame_counter;
   iface->finish_frame = meta_stage_native_finish_frame;
+  iface->get_update_time = meta_stage_native_get_update_time;
 }
diff --git a/src/backends/x11/meta-stage-x11.c b/src/backends/x11/meta-stage-x11.c
index 77e96f97e..7331bc586 100644
--- a/src/backends/x11/meta-stage-x11.c
+++ b/src/backends/x11/meta-stage-x11.c
@@ -488,6 +488,17 @@ meta_stage_x11_get_frame_counter (ClutterStageWindow *stage_window)
   return cogl_onscreen_get_frame_counter (stage_x11->onscreen);
 }
 
+static gint64
+meta_stage_x11_get_update_time (ClutterStageWindow *stage_window)
+{
+  ClutterStageCogl *stage_cogl = CLUTTER_STAGE_COGL (stage_window);
+
+  if (stage_cogl->pending_swaps)
+    return -1; /* in the future, indefinite */
+
+  return stage_cogl->update_time;
+}
+
 static void
 meta_stage_x11_finalize (GObject *object)
 {
@@ -549,6 +560,7 @@ clutter_stage_window_iface_init (ClutterStageWindowInterface *iface)
   iface->can_clip_redraws = meta_stage_x11_can_clip_redraws;
   iface->get_views = meta_stage_x11_get_views;
   iface->get_frame_counter = meta_stage_x11_get_frame_counter;
+  iface->get_update_time = meta_stage_x11_get_update_time;
 }
 
 static inline void
-- 
2.26.2

